### PR TITLE
[ShapeUp] feat: add secrets

### LINKED ISSUE #3100: [ShapeUp] Add secrets to sessions
This [ShapeUp project](https://www.notion.so/Bringing-secrets-to-sessions-7b086846b5ab4d58bdc064342d39c528) aims to add secrets to user sessions. There are 2 changes in the UI:

* Add a page to view/manage user secrets.
* Allow users to inject secrets into sessions as environmental variables or files mounted through k8s secrets.

_Mind that this won't affect sessions 2.0_

### PR DESCRIPTION
**Pull request by lorenzo-cavazzi**
Secrets in the UI -- feature branch

/deploy renku=release-0.52.x

fix #3100

### CODE REVIEW

File: client/src/features/secrets/SecretNew.tsx [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+                  autoComplete="off"
+                  className={cx(
+                    "form-control",
+                    "rounded-0",
+                    "rounded-start",
+                    !showPlainText && styles.blurredText,
+                    errors.value && "is-invalid"
+                  )}
+                  id="new-secret-value"
+                  type="textarea"
```
Comment (ciyer) [inline client/src/features/secrets/SecretNew.tsx:163]:
```suggestion
                  id="new-secret-value"
                  spellCheck="false"
                  type="textarea"
```

This field should probably not be spellchecked.

image

File: client/src/features/secrets/SecretEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                      autoComplete="off"
+                      className={cx(
+                        "form-control",
+                        "rounded-0",
+                        "rounded-start",
+                        !showPlainText && styles.blurredText,
+                        errors.value && "is-invalid"
+                      )}
+                      id="edit-secret-value"
+                      type="textarea"
```
Comment (ciyer) [inline client/src/features/secrets/SecretEdit.tsx:141]:
```suggestion
                      id="edit-secret-value"
                      spellCheck="false"
                      type="textarea"
```

**Review by ciyer — CHANGES_REQUESTED** [review]
Looks very good and works well! I have a small suggestion to turn off spellcheck in the secrets input.

**Comment by lorenzo-cavazzi** [comment]
@ciyer  I added your suggestions to the code, although not all browsers consider that option
image

**Review by ciyer — APPROVED** [review]
LGTM!

**Comment by leafty** [comment]
> @ciyer I added your suggestions to the code, although not all browsers consider that option

Note: It looks like it's added from a chrome extension, not the base browser itself.

**Comment by lorenzo-cavazzi** [comment]
> Note: It looks like it's added from a chrome extension, not the base browser itself.

@leafty That's actually the Firefox browser spellcheck.

On top of that, I also have a browser extension that would take priority and ofc ignore the setting as well :laughing: 

The additional icon on the right side of the field was misleading, it was temporarily off. It shows slightly differently when it's enabled on the page and it finds errors.

image

**Pull request merged**